web/src/pages/api/public/metrics/daily.ts

```diff
@@ -47,53 +53,44 @@ export default withMiddlewares({
               SUM(o.completion_tokens) "outputUsage",
               SUM(o.total_tokens) "totalUsage",
               COALESCE(SUM(o.calculated_total_cost), 0)::DOUBLE PRECISION as "totalCost"
-            FROM
-              traces t
-            LEFT JOIN observations_view o ON o.trace_id = t.id AND o.project_id = t.project_id
+            FROM traces t
+            LEFT JOIN observations_view o
+            ON o.trace_id = t.id
+            AND o.project_id = t.project_id
             WHERE o.start_time IS NOT NULL
               AND t.project_id = ${auth.scope.projectId}
               ${traceNameCondition}
               ${userCondition}
               ${tagsCondition}
               ${fromTimestampCondition}
               ${toTimestampCondition}
-            GROUP BY
-              1,
-              2
-            ORDER BY
-              1,
-              2
+              ${fromObservationStartTimeCondition}
+              ${toObservationStartTimeCondition}
```

Comment:
@marcklingen My understanding from our documentation is that traces should wrap all observations, i.e. my understanding is that t.timestamp <= o.start_time. Could you confirm this?

In addition, I updated the fern doc to make this change here explicit.

I would like to add the observation.start_time condition, because it allows us to use the observation indexes more effectively. Without it, we seem to join _all_ observations for the given project whereas this flag allows for some pre-filtering.

Reply:
@marcklingen Yeah, it seems to perform a nested loop join and therefore only uses the trace_id later in the flow. Adding the start_time makes it use the trace_id_project_id_start_time index which seems more performant.